### PR TITLE
test: switch InteractiveStatmentExecutorTest to use mockito

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 
What the title says
### Testing done 
Test only change
### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java [hunk 1]
```diff
@@ -730,9 +699,6 @@ public void shouldRestoreLegacyRunScriptCommand() {
             0L
         )
     );
-
-    // Then:
-    verify(mockParser, mockEngine, mockQuery);
```
Comment (vcrfxia) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: Can we verify the `mockEngine.execute()` call expected in `mockReplayRunScript()`? That seems to be what this test was meant to check.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java [hunk 2]
```diff
@@ -406,6 +375,50 @@ public void shouldHandlePriorStatements() {
     Assert.assertEquals(CommandStatus.Status.ERROR, statusStore.get(ctasCommandId).getStatus());
   }
 
+  private Query mockCSASQuery() {
+    final Query mockQuery = mock(Query.class);
+    when(mockQuery.getLimit()).thenReturn(OptionalInt.empty());
+    return mockQuery;
+  }
+  //
```
Comment (vcrfxia) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: nit

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java [hunk 3]
```diff
@@ -527,42 +496,42 @@ public void shouldSkipStartWhenReplayingLog() {
...
         )
     );
 
     // Then:
-    verify(mockParser, mockEngine, mockQuery);
+    verify(mockQueryIdGenerator, times(1)).activateNewGenerator(anyLong());
   }
 
   @Test
+  @SuppressWarnings("unchecked")
```
Comment (vcrfxia) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: Out of curiosity, why was this added?

Comment (stevenpyzhang) [reply ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: Datasource takes a generic type and mockito yells at me when mocking generics. I haven't found a really good way around this so far.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java (hunk 1 above)
Comment (stevenpyzhang) [reply ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: I added a verify that the query isn't started. I think that's what should be tested since the only difference between `shouldRestoreLegacyRunScriptCommand` and `shouldHandleLegacyRunScriptCommand` is the mode, which causes `handleLegacyRunScript` in the executor to act differently.

**Comment by stevenpyzhang** [comment]
@vcrfxia I added Given/When/Then to most of the tests. I'm not seeing any additional mock functionality that could be moved to setup() though. And the large chunks of mock specific to some tests...Not sure the best way to break it up into smaller chunks, it doesn't seem too difficult to understand.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java [hunk 4]
```diff
@@ -98,23 +88,27 @@
...
+public class InteractiveStatementExecutorTest {
 
   private static final Map<String, String> PRE_VERSION_5_NULL_ORIGINAL_PROPS = null;
 
   private KsqlEngine ksqlEngine;
   private InteractiveStatementExecutor statementExecutor;
   private KsqlConfig ksqlConfig;
 
-  private final StatementParser mockParser = niceMock(StatementParser.class);
+  private final StatementParser mockParser = mock(StatementParser.class);
```
Comment (agavra) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: nit: if we're following the pattern this should `@RunWith(MockitoJUnitRunner.class)` and these should use the `@Mock` annotation

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java [hunk 5]
```diff
@@ -279,47 +272,23 @@ public void shouldBuildQueriesWithPersistedConfig() {
...
-      buffer.append("status(").append(status).append(")");
-    }
-  }
-
-  private static CommandStatus sameStatus(final CommandStatus.Status status) {
-    reportMatcher(new StatusMatcher(status));
-    return null;
+    
+    // Then:
+    verify(mockQueryMetadata, times(1)).start();
```
Comment (agavra) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: strictly speaking, we're changing this test. It used to verify also that all of the other mocks happened exactly once (e.g. that `plan` was called, that `execute` was called and that `start` was called). I think this coverage is actually important since it helped me find a bug when I was doign #3933

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java [hunk 6]
```diff
@@ -574,36 +556,44 @@ public void shouldCascade4Dot1DropStreamCommand() {
     );
 
     // Then:
-    verify(mockParser, mockEngine, mockMetaStore);
+    verify(mockQueryMetadata, times(1)).close();
```
Comment (agavra) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: same comment as above

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java (hunk 4 above)
Comment (stevenpyzhang) [reply ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: Added

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java (hunk 5 above)
Comment (stevenpyzhang) [reply ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutorTest.java]: Discussed a bit, while it would be better to have explicit verification, Mockito will throw unnecessary stubbing exceptions if a mock isn't used so in a sense it "verifies" that the mock was called with the exact params. (Don't silence those exceptions)
